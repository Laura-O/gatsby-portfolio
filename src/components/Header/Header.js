import React, { Component } from 'react';
import styled from 'styled-components';
import Menu from '../Menu/Menu';
import * as constants from '../../layouts/shared/style-constants';

const HeaderWrapper = styled.header`
  display: flex;
  justify-content: flex-end;
  background: ${constants.darkBackground};
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  padding: 1.5rem;
  height: ${constants.headerHeight};
  z-index: ${constants.zIndexHeader};
`;

const Hamburger = styled.div`
  width: 30px;
  height: 25px;
  position: relative;
  transform: rotate(0deg);
  cursor: pointer;

  span {
    display: block;
    position: absolute;
    height: 20%;
    width: 100%;
    background: white;
    opacity: 1;
    left: 0;
    transform: rotate(0deg);
    transition: 0.4s cubic-bezier(0.18, 0.89, 0.32, 1.1);
    z-index: 3;

    &:nth-child(1) {
      top: 0;
    }

    &:nth-child(2),
    &:nth-child(3) {
      top: 40%;
    }

    &:nth-child(4) {
      top: 80%;
    }
  }
  ${props => {
    if (props.active) {
      return `
      z-index: ${constants.zIndexHamburger};
    
      span {
        &:nth-child(1) {
          top: 18px;
          width: 0%;
          left: 50%;
        }
    
        &:nth-child(2) {
          transform: rotate(45deg);
        }
    
        &:nth-child(3) {
          transform: rotate(-45deg);
        }
    
        &:nth-child(4) {
          top: 18px;
          width: 0%;
          left: 50%;
        }
      }
      `;
    }
    return null;
  }};
`;

export default class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      menuActive: false,
    };

    this.toggleMenu = this.toggleMenu.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }

  componentDidMount() {
    document.addEventListener('keydown', this.handleKeyPress);
  }

  toggleMenu() {
    this.setState({ menuActive: !this.state.menuActive });
  }

  handleKeyPress(event) {
    if (event.keyCode === 27) {
      this.toggleMenu(false);
    }
  }
  render() {
    const { menuActive } = this.state;

    return (
      <HeaderWrapper>
        <Hamburger
          active={menuActive}
          onClick={() => this.toggleMenu()}
          className="hamburger"
        >
          <span />
          <span />
          <span />
          <span />
        </Hamburger>

        <Menu toggleNav={() => this.toggleMenu()} active={menuActive} />
      </HeaderWrapper>
    );
  }
}
