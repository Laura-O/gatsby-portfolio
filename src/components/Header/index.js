import React, { Component } from 'react'
import Link from 'gatsby-link'
import classNames from 'classnames'
import Menu from '../Menu'
import styles from './styles.module.scss'

export default class Header extends Component {
  constructor(props) {
    super(props)
    this.state = {
      menuActive: false,
    }

    this.toggleMenu = this.toggleMenu.bind(this)
    this.handleKeyPress = this.handleKeyPress.bind(this)
  }

  componentDidMount() {
    document.addEventListener('keydown', this.handleKeyPress)
  }

  toggleMenu() {
    console.log(this.state.menuActive)
    this.setState({ menuActive: !this.state.menuActive })
  }

  handleKeyPress(event) {
    if (event.keyCode === 27) {
      this.toggleMenu(false)
    }
  }
  render() {
    const { menuActive } = this.state

    const barStyle = this.state.menuActive ? 'animate' : ''
    console.log(barStyle)

    return (
      <header className={styles.header}>
        <div
          className={menuActive ? styles.active : styles.icon}
          onClick={() => this.toggleMenu()}
        >
          <span />
          <span />
          <span />
          <span />
        </div>
        <Menu toggleNav={() => this.toggleMenu()} active={menuActive} />
      </header>
    )
  }
}
