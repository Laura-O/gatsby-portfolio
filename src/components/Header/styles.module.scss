@import '../../scss/variables.scss';

$bar-width: 50px;
$bar-height: 5px;
$bar-spacing: 12px;

.header {
  display: flex;
  justify-content: flex-end;
  background: $colorBlack;
  color: $colorWhite;
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  padding: 5px;
  height: $headerHeight;
  z-index: $zIndexHeader;
}

.hamburger {
  position: relative;
  z-index: $zIndexHamburger;
}

.container {
  padding: 10px;
}

.icon {
  width: 28px;
  height: 22px;
  position: relative;
  transform: rotate(0deg);
  cursor: pointer;
}

.icon span {
  display: block;
  position: absolute;
  height: 20%;
  width: 100%;
  background: $colorWhite;
  opacity: 1;
  left: 0;
  transform: rotate(0deg);
  transition: 0.25s cubic-bezier(0.18, 0.89, 0.32, 1.28);
  z-index: $zIndexHamburger;
}

.icon span:nth-child(1) {
  top: 0;
}

.icon span:nth-child(2),
.icon span:nth-child(3) {
  top: 40%;
}

.icon span:nth-child(4) {
  top: 80%;
}

.active {
  composes: icon;
  z-index: $zIndexHamburger;
}

.active span:nth-child(1) {
  top: 18px;
  width: 0%;
  left: 50%;
}

.active span:nth-child(2) {
  transform: rotate(45deg);
}

.active span:nth-child(3) {
  transform: rotate(-45deg);
}

.active span:nth-child(4) {
  top: 18px;
  width: 0%;
  left: 50%;
}
