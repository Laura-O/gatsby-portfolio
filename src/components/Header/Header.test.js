import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />);
    expect(wrapper.length).toBe(1);
  });

  test('rendering', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('keypress', () => {
    const instance = wrapper.instance();
    wrapper.setState({ menuActive: false });
    instance.handleKeyPress({ keyCode: 27 });
    expect(wrapper.state().menuActive).toBe(true);
  });
});
