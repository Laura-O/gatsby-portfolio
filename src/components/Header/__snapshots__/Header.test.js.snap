// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div
          active={false}
          className="hamburger"
          onClick={[Function]}
        >
          <span />
          <span />
          <span />
          <span />
        </styled.div>,
        <Menu
          active={false}
          toggleNav={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "children": Array [
            <span />,
            <span />,
            <span />,
            <span />,
          ],
          "className": "hamburger",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": false,
          "toggleNav": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div
            active={false}
            className="hamburger"
            onClick={[Function]}
          >
            <span />
            <span />
            <span />
            <span />
          </styled.div>,
          <Menu
            active={false}
            toggleNav={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": Array [
              <span />,
              <span />,
              <span />,
              <span />,
            ],
            "className": "hamburger",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "toggleNav": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
